src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1147,6 +1197,15 @@ private AuthenticationResult acquireTokenAfterValidation(CallbackHandler callbac
             AuthenticationResult result = null;
             request.setVersion(getVersionName());
             request.setBrokerAccountName(request.getLoginHint());
+            
+            //check if App gives permission to AccountManager
+            List<String> permissionCheckResult = verifyManifestPermissions();
```

Comment:
Right now the name and the purpose of the method is different. It checks for missing permissions and returns a list. It may as well be named as getMissingPermissions. I think current name is good and verify should itself throw exception if there are missing permissions.